src/libraries/System.IO.MemoryMappedFiles/src/System/IO/MemoryMappedFiles/MemoryMappedFile.Unix.cs

```diff
@@ -238,13 +245,23 @@ private static SafeFileHandle CreateSharedBackingObject(Interop.Sys.MemoryMapped
 
             static string GenerateMapName()
             {
-                const int MaxSharedMemoryObjectNameLength = 32; // SHM_NAME_MAX on OSX ARM64, on other systems it's equal PATH_MAX (250)
-                // The POSIX shared memory object name must begin with '/'.  After that we just want something short (32) and unique.
-                return string.Create(MaxSharedMemoryObjectNameLength, 0, (span, state) =>
+                // macOS shm_open documentation says that the sys-call can fail with ENAMETOOLONG if the name exceeds SHM_NAME_MAX characters.
+                // The problem is that SHM_NAME_MAX is not defined anywhere and is not consistent amongst macOS versions (arm64 vs x64 for example).
+                // It was reported in 2008 (https://lists.apple.com/archives/xcode-users/2008/Apr/msg00523.html),
+                // but considered to be by design (http://web.archive.org/web/20140109200632/http://lists.apple.com/archives/darwin-development/2003/Mar/msg00244.html).
+                // According to https://github.com/qt/qtbase/blob/1ed449e168af133184633d174fd7339a13d1d595/src/corelib/kernel/qsharedmemory.cpp#L53-L56 the actual value is 30.
+                // Some other OSS libs use 32 (we did as well, but it was not enough) or 31, but we prefer 30 just to be extra safe.
+                const int MaxNameLength = 30;
+                // The POSIX shared memory object name must begin with '/'.  After that we just want something short (30) and unique.
+                const string NamePrefix = "/dotnet_";
+                return string.Create(MaxNameLength, 0, (span, state) =>
                 {
-                    Guid.NewGuid().TryFormat(span, out int charsWritten, "N");
-                    Debug.Assert(charsWritten == MaxSharedMemoryObjectNameLength);
-                    "/dotnet_".CopyTo(span);
+                    Span<char> guid = stackalloc char[32];
+                    Guid.NewGuid().TryFormat(guid, out int charsWritten, "N");
+                    Debug.Assert(charsWritten == 32);
+                    NamePrefix.CopyTo(span);
+                    guid.Slice(0, MaxNameLength - NamePrefix.Length).CopyTo(span.Slice(NamePrefix.Length));
+                    Debug.Assert(Encoding.UTF8.GetByteCount(span) <= MaxNameLength); // the standard uses Utf8
```

Comment:
You could also do something like:
```C#
const int MaxNameLength = 30;
const string NamePrefix = "/dotnet_";
return string.Create(MaxNameLength, 0, (span, state) =>
{
    NamePrefix.CopyTo(span);
    span = span.Slice(NamePrefix.Length);

    Span<byte> chars = stackalloc byte[MaxNameLength - NamePrefix.Length];
    RandomNumberGenerator.Fill(chars);

    string values = "0123456789abcdefghijklmnopqrstuvwxyz";
    for (int i = 0; i < span.Length; i++)
    {
        span[i] = values[chars[i] % values.Length];
    }
});
```

Reply:
@stephentoub I could, but it would require new dependency?

Also I believe that we could generate the name in the native code, as all we need to do is: generate the name, call `shm_open` and call `shm_unlink` right after that

https://github.com/dotnet/runtime/blob/a24364a09d9aea98b545f16689a53bafc6b18c14/src/libraries/System.IO.MemoryMappedFiles/src/System/IO/MemoryMappedFiles/MemoryMappedFile.Unix.cs#L213-L216

Reply:
@stephentoub I would prefer not to add a new dependency. What is your opinion on that?

Reply:
I don't think another dependency here is a big deal (they're all within the shared framework), but I'm fine with you sticking with what you have.